1.21.5

onebeastchris opened the pull request.

basaigh committed:
25w02a

basaigh committed:
25w03a

basaigh committed:
25w04a (hopefully)

basaigh committed:
25w05a

basaigh committed:
25w06a

basaigh committed:
Add some missing changes from 25w03/4a

basaigh committed:
25w07a

basaigh committed:
25w08a

basaigh committed:
Fix test

basaigh committed:
1.21.5-pre1

basaigh committed:
1.21.5-rc1

basaigh committed:
Fix tests again

basaigh committed:
Ensure components and metadata is correct

basaigh committed:
Update to reflect the use of fixed length long arrays in data palettes.

eclipseisoffline committed:
Fix data palette reading/writing (#891)

basaigh committed:
Target 1.21.5 release protocol

basaigh committed:
Correctly read exact component matchers in villager trades

basaigh committed:
Fix ChickenVariant type and correctly encode untrusted data components

basaigh committed:
Allow for unexpected `CommandType` values as vanilla does

basaigh committed:
Add missing argument type

AlexProgrammerDE commented:
This will not be a very useful release for anyone except Geyser if it's missing data component hashing. Will Geyser offer an item hashing implementation or render this library useless for inventory interactions?

onebeastchris committed:
Fix unit reader/writer for the INTANGIBLE_PROJECTILE component

basaigh committed:
Fix `Can't join: ByteToMessageCodec$1.decode() did not read anything but decoded a message` (#893)

onebeastchris committed:
Fix EquipmentSlot reader in the Equippable item component

eclipseisoffline committed:
Data components map improvements (#894)

* Improve DataComponents class

- Added #contains and #remove methods
- Make #put allow null values
- Add Javadoc to make usage of null values clear

* Add fromKey method to DataComponentTypes

onebeastchris committed:
Add BuiltinFlags.ALLOCATOR

eclipseisoffline committed:
Properly put null values in data component map (#895)

onebeastchris committed:
Support netty 4.2.1, add system property for opt-in io_uring (#897)

* Support netty 4.2.1, add system property for opt-in io_uring

* check for io.netty.channel.MultiThreadIoEventLoopGroup instead

* use Boolean.getBoolean instead
